src/transformers/trainer.py

```diff
@@ -1972,8 +1972,18 @@ def train(
                 self._move_model_to_device(self.model, args.device)
             self.model_wrapped = self.model
 
+
+        def reduce_batch_size_fn_wrapper(train_batch_size, args):
+            effective_batch_size = args.gradient_accumulation_steps * train_batch_size
+            def reduce_batch_size_fn():
+                nonlocal train_batch_size
+                train_batch_size = train_batch_size // 2
+                args.gradient_accumulation_steps = effective_batch_size // train_batch_size
+                return train_batch_size
+            return reduce_batch_size_fn
+
         inner_training_loop = find_executable_batch_size(
```

Comment:
The `trainer_utils`'s `find_executable_batch_size` will then also need changing.

Also need to be careful about versioning inside that func

Reply:
It should be set as `>=0.34.0`

Reply:
sorry, I forgot to commit a file. should be there now